packages/framework/request-handler/src/requestHandlers.ts

```diff
@@ -40,7 +40,7 @@ export const rootDataStoreRequestHandler = async (request: IRequest, runtime: IC
         const rootDataStore = await runtime.getRootDataStore(id, wait);
         return rootDataStore.IFluidRouter.request(requestParser.createSubRequest(1));
     } catch (error) {
-        return create404Response(request);
```

Comment:
this doesn't seem right. it should only return undefined if there in no root data store

Reply:
i think getRootDataStore return undefined if the id doesn't exist, that's the only case we should return undefined. in the error case we probably want to return a 500.

Reply:
uggh. looked at the code. looks like getRootDataStore fails if the data store does not exist. but to try catching getRootDataStore and returning undefined is ok, but we shouldn't wrap `rootDataStore.IFluidRouter.request(requestParser.createSubRequest(1));` as if that throws, the request was to a valid datastore, it just failed to process it. in that case we should return 500

Reply:
yep. I'll have a PR out for this change.

Reply:
Agree with @anthony-murphy here on using 500. I'd leave 404 only for builder as a means to say "I tried all the handlers, none of them resolved."

Reply:
try/catch PR for `rootDataStoreRequestHandler`: https://github.com/microsoft/FluidFramework/pull/8742